### PR TITLE
Add hmac signing to storm (SYN-4305, SYN-4325)

### PR DESCRIPTION
**Pull request by vEpiphyte**

### CODE REVIEW

File: synapse/tests/test_lib_stormlib_hashes.py [hunk 1]
```diff
@@ -22,3 +26,111 @@ async def test_stormlib_crypto_hashes(self):
                 digests = {k: binascii.hexlify(v).decode() for k, v in hashset.digests()}
 
                 self.eq(hashes, digests)
+
+    async def test_stormlib_crypto_hmac(self):
+
+        async with self.getTestCore() as core:
+
+            q = '''$key=$key.encode() $mesg=$mesg.encode()
+            return ( $lib.crypto.hmac.sign(key=$key, mesg=$mesg, digest=$mode) )
```
Comment (vEpiphyte) [inline synapse/tests/test_lib_stormlib_hashes.py:35]: I kind of think this may be better suited as ``$lib.crypto.hashes.hmac()`` ?

Comment (Cisphyx) [reply synapse/tests/test_lib_stormlib_hashes.py:35]: Yeah I think that fits in better

Comment (invisig0th) [reply synapse/tests/test_lib_stormlib_hashes.py:35]: I def would not consider HMAC "a hash algo" though. We should probably discuss.

File: synapse/lib/stormlib/hashes.py [hunk 2]
```diff
@@ -54,3 +57,51 @@ async def _sha256(self, byts):
...
         return hashlib.sha512(byts).hexdigest()
+
+@s_stormtypes.registry.registerLib
+class LibHmac(s_stormtypes.Lib):
+    '''
+    A Storm library for computing RFC2104 HMAC values.
+    '''
+    _storm_locals = (
+        {'name': 'digest', 'desc': '''
+        Compute the digest value of a message using RFC2140 HMAC.
```
Comment (Cisphyx) [inline synapse/lib/stormlib/hashes.py:68]: should be RFC2104

File: synapse/lib/stormlib/hashes.py [hunk 3]
```diff
@@ -54,3 +57,51 @@ async def _sha256(self, byts):
...
+
+        Examples:
+            Compute the HMAC-SHA256 digest for a message with a secret key::
+
+                $digest = $lib.crypto.hmac.digest(key=$secretKey.encode(), mesg=$mesg.encode())
+        ''',
+         'type': {'type': 'function', '_funcname': '_digest',
+                  'args': (
+                      {'name': 'key', 'type': 'bytes', 'desc': 'The key to use for the HMAC calculation.'},
+                      {'name': 'mesg', 'type': 'bytes', 'desc': 'The message use for the HMAC calculation.'},
```
Comment (Cisphyx) [inline synapse/lib/stormlib/hashes.py:78]: message "to" use

File: synapse/lib/stormlib/hashes.py (hunk 2 above)
Comment (invisig0th) [reply synapse/lib/stormlib/hashes.py:68]: nice catch!

**Review by Cisphyx — APPROVED** [review]

**Pull request merged**